BT-8129: Pass cookie, allocator, and deallocator to ubpf_create

kenchiang opened the pull request:
Reviewer: @3mp4y5 
Pass zmalloc_cookie and zmalloc and free functions to ubpf_create, used for allocation and deallocation

kenchiang committed:
BT-8129
Pass zmalloc_cookie and zmalloc and free functions to ubpf_create,
used for allocation and deallocation

3mp4y5 commented on vm/ubpf_vm.c:
```diff
@@ -45,20 +45,28 @@ void ubpf_set_error_print(struct ubpf_vm *vm, int (*error_printf)(FILE* stream,
 
-    vm->ext_funcs = calloc(MAX_EXT_FUNCS, sizeof(*vm->ext_funcs));
+    vm->zmalloc_cookie = zmalloc_cookie;
+    vm->zmalloc_fn = zmalloc_fn;
```
Why not create a few sample zmalloc_fn's, and if the passed in function pointers are zero, then use the local functions (which could use calloc/free for example)

I'm thinking we're the only user ... but just in case, do you think it makes sense to create a ubpf_create_XXX() which has the zmalloc/free fn's, and leave the old ubpf_create() as is, and use that to call ubpf_create_XXX(...) with zero's, which would cause it to use the local versions of the functions?

kenchiang replied:
Reintroduced ubpf_create() which uses default allocator and deallocator functions.

3mp4y5 reviewed (commented):
a few Q's

kenchiang committed:
BT-8129
Reintroduce ubpf_create(), which calls ubpf_create_ext() with default allocator and deallocator functions

3mp4y5 reviewed (approved):
LGTM
